src/zoid/buttons/util.js

```diff
@@ -357,3 +362,60 @@ export function getButtonSize(
     }
   }
 }
+
+export const getModal: (
+  clientID: string,
+  merchantID: $ReadOnlyArray<string> | void
+) => Object = memoize(async (clientID, merchantID) => {
+  try {
+    const namespace = getNamespace();
+    if (!window[namespace]?.MessagesModal) {
+      const modalBundleUrl = () => {
+        let envPiece;
+        switch (getEnv()) {
+          case "local":
+          case "test":
+          case "stage":
+            envPiece = "stage";
+            break;
+          case "sandbox":
+            envPiece = "sandbox";
+            break;
+          case "production":
+          default:
+            envPiece = "js";
+        }
+        return `https://www.paypalobjects.com/upstream/bizcomponents/${envPiece}/modal.js`;
+      };
+
+      // eslint-disable-next-line no-restricted-globals
+      await new Promise((resolve, reject) => {
+        const script = document.createElement("script");
+        script.src = modalBundleUrl();
+        script.setAttribute("data-pp-namespace", namespace);
+        script.addEventListener("error", (err: Event) => {
+          reject(err);
+        });
+        script.addEventListener("load", () => {
+          document.body?.removeChild(script);
+          resolve();
+        });
+        document.body?.appendChild(script);
+      });
+    }
+
+    const modal = window[namespace].MessagesModal;
+    return modal({
```

Comment:
Since this variable isn't really being reused are we able to combine these to `window[namespace].MessagesModal({ ... })` since that the common form we're used to seeing when creating SDK components?